processors/base/helpers/pybossa_tasks_updater.py

```diff
@@ -0,0 +1,113 @@
+# -*- coding: utf-8 -*-
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+import logging
+import time
+import pbclient
+logger = logging.getLogger(__name__)
+
+
+class PyBossaTasksUpdater(object):
+    def __init__(self, endpoint, api_key, project_id):
+        pbclient.set('endpoint', endpoint)
+        pbclient.set('api_key', api_key)
+        self.project_id = project_id
+
+    def run(self, tasks_data, unique_keys):
+        add, delete, update = self._filter_tasks(tasks_data, unique_keys)
+
+        msg = '{} tasks to add, {} tasks to delete, {} tasks to update'
+        logger.debug(msg.format(len(add), len(delete), len(update)))
+
+        for task_data in add:
+            self._add_task(task_data)
+
+        for task in delete:
+            self._delete_task(task)
+
+        for task in update:
+            self._update_task(task)
+
+    def _filter_tasks(self, tasks_data, unique_keys):
+        get_key = lambda value: tuple((key, value[key]) for key in unique_keys) # noqa: E731
+        get_keys = lambda values: [get_key(value) for value in values] # noqa: E731
+
+        existing_tasks = self._get_existing_tasks()
+        existing_tasks_info = [task.data['info'] for task in existing_tasks]
+        existing_ids = get_keys(existing_tasks_info)
+        new_ids = get_keys(tasks_data)
+
+        tasks_to_add = [task_data for task_data in tasks_data
+                        if get_key(task_data) not in existing_ids]
+        tasks_to_delete = [existing_tasks[i]
+                           for i, task_info in enumerate(existing_tasks_info)
+                           if get_key(task_info) not in new_ids]
+
+        tasks_to_update = []
```

Comment:
Are we allowed to update existing tasks? Even if they already have received answers?